proxy/src/serve_tasks.rs

```diff
@@ -420,7 +375,7 @@ pub(crate) async fn validate_and_decrypt(json: Value) -> Result<Value, SamplyBea
     if let Value::Array(arr) = json {
         let mut results = Vec::with_capacity(arr.len());
         for value in arr {
-            results.push(validate_and_decrypt(value).await?);
+            results.push(Box::pin(validate_and_decrypt(value)).await?);
```

Comment:
Because of async recursion.
Remember that Rust futures compile to perfectly sized state machines which means for a recursive function it would create an infinity sized future which is why we need the heap allocation.
Before we used a crate for that which did this but more complicated.
But since recent rust versions you can just do this.

Reply:
I fully understand the heap allocation. In my understanding, pinning guarantees an unmovable address on the heap. I'm not sure why this is needed. But this is mere curiosity :)

Reply:
Well all futures need to be pinned either on the stack or heap to guarantee that none of the pointers change during polling. This is why the `Future` trait takes a `Pin<&mut Self>` and this is also why we need to pin the boxed future because otherwise it would not implement `Future` anymore. See [here](https://doc.rust-lang.org/std/future/trait.Future.html#impl-Future-for-Box%3CF,+A%3E)

Reply:
Thanks!